lib/library/AssignLibraryPlugin.js

```diff
@@ -222,8 +225,12 @@ class AssignLibraryPlugin extends AbstractLibraryPlugin {
 	 * @param {LibraryContext<T>} libraryContext context
 	 * @returns {string | undefined} bailout reason
 	 */
-	embedInRuntimeBailout(module, { chunk }, { options, compilation }) {
-		const topLevelDeclarations =
+	embedInRuntimeBailout(
+		module,
+		{ chunk, codeGenerationResults },
+		{ options, compilation }
+	) {
+		let topLevelDeclarations =
 			module.buildInfo && module.buildInfo.topLevelDeclarations;
```

Comment:
> So the topLevelDeclarations from code generations overrides the original topLevelDeclarations, since it's basically a better version of it. It includes renames, which means some original topLevelDeclarations might be no longer in the code generation topLevelDeclarations, e. g. all reserved names.

yeah, reserved names (globals) is a case.

> Assuming it should also be possible that buildInfo.topLevelDeclarations is not set but data.topLevelDeclarations

not sure that this wont be a breaking change.. I have change hooks api ( added `codeGenerationResults` )

Reply:
btw case 
```js
{
		entry: {
			index: path.resolve(__dirname, "./index.js")
		},
		output: {
			library: "setTimeout",
			libraryExport: "default"
		},
		optimization: {
			concatenateModules: true
		}
	}
```
will not work.. since "global" `setTimeout` points to variable